Library/include/CSP/Multiplayer/Components/TextSpaceComponent.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// @file TextSpaceComponent.h
+/// @brief Definitions and support for text components.
+
+#pragma once
+
+#include "CSP/CSPCommon.h"
+#include "CSP/Common/String.h"
+#include "CSP/Multiplayer/ComponentBase.h"
+#include "CSP/Multiplayer/Components/Interfaces/ITransformComponent.h"
+#include "CSP/Multiplayer/Components/Interfaces/IVisibleComponent.h"
+
+
+namespace csp::multiplayer
+{
+
+/// @brief Enumerates the list of properties that can be replicated for an text space component.
+enum class TextPropertyKeys
+{
+	Position,
+	Rotation,
+	Scale,
+	Text,
+	TextColor,
+	BackgroundColor,
+	IsBackgroundVisible,
+	Width,
+	Height,
+	BillboardMode,
+	IsVisible,
+	IsARVisible,
+	Num
+};
+
+
+/// @brief The billboard mode supported by this text space component.
+enum class TextSpaceComponentBillboardMode
```

Comment:
Should we refactor [this enum](https://github.com/magnopus-opensource/connected-spaces-platform/blob/develop/Library/include/CSP/Multiplayer/Components/ImageSpaceComponent.h#L51) so that it becomes shared with image components and text components, rather than introducing a new one with the same values?

Reply:
I wasn't sure what to do about this, as there is indeed a BillboardMode enum class elsewhere. Clean code practices dictate promotion to it's own file, but I was trying to exercise a light touch on this ticket. If the consensus is to refactor, I am happy to do it.

Reply:
Promoting to its own file makes sense to me!

Reply:
Changes in latest look great, thanks!